### PR TITLE
IOSS: Add a Faodel DatabaseIO component

### PR DESCRIPTION
**Pull request by tkordenbrock**
This PR replaces the DataWarehouse DatabaseIO component with a Faodel DatabaseIO component. This component enables the transfer of CGNS and Exodus data to a Faodel Distributed Hash Table.

@gjtempl @craigulmer @scottnlevy

### CODE REVIEW

File: install-tpl.sh [hunk 1]
```diff
@@ -95,6 +95,9 @@ KOKKOS=${KOKKOS:-NO}
 KOKKOS=`check_valid KOKKOS`
 
 H5VERSION=${H5VERSION:-V110}
+FAODEL=${FAODEL:-OFF}
```
Comment (gsjaardema) [inline install-tpl.sh:98]: I think these should be NO instead of OFF

File: install-tpl.sh [hunk 2]
```diff
@@ -95,6 +95,9 @@ KOKKOS=${KOKKOS:-NO}
...
+ADIOS2=${ADIOS2:-OFF}
```
Comment (gsjaardema) [inline install-tpl.sh:99]: Duplicated line with below.

File: cmake/ProjectDependenciesSetup.cmake [hunk 3]
```diff
@@ -0,0 +1,40 @@
+function(disable_if_no_fortran package)
```
Comment (gsjaardema) [inline cmake/ProjectDependenciesSetup.cmake:1]: Is this somehow now pulled into the CMakeLists.txt from where it was removed?

File: packages/seacas/libraries/exodus/CMakeLists.txt [hunk 4]
```diff
@@ -35,6 +35,12 @@ TRIBITS_ADD_LIBRARY(
 )
 
 set_property(TARGET exodus PROPERTY C_STANDARD 99)
+TARGET_INCLUDE_DIRECTORIES(
```
Comment (gsjaardema) [inline packages/seacas/libraries/exodus/CMakeLists.txt:38]: Why were these added?  I think the include files have been getting installed without these...

File: packages/seacas/libraries/ioss/src/CMakeLists.txt [hunk 5]
```diff
@@ -74,6 +74,15 @@ IF (MINGW)
   target_link_libraries(Ioss psapi)
 ENDIF ()
 
+TARGET_INCLUDE_DIRECTORIES(
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/CMakeLists.txt:77]: Is this needed?

File: packages/seacas/libraries/ioss/src/Ioss_CodeTypes.h [hunk 6]
```diff
@@ -37,9 +37,9 @@ inline const std::string IOSS_SYM_TENSOR() { return std::string("sym_tensor_33")
 #define SEACAS_HAVE_MPI
 /* #undef IOSS_THREADSAFE */
 /* #undef SEACAS_HAVE_KOKKOS */
-/* #undef SEACAS_HAVE_DATAWAREHOUSE */
 #define SEACAS_HAVE_EXODUS
 #define SEACAS_HAVE_CGNS
+#define SEACAS_HAVE_FAODEL
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_CodeTypes.h:42]: This block is if we have FAODEL activated in Sierra.  This should not be defined until FAODEL is added as a TPL in Sierra

File: packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.C [hunk 7]
```diff
@@ -23,4 +23,37 @@ namespace Ioss {
   const double *CoordinateFrame::origin() const { return &pointList_[0]; }
   const double *CoordinateFrame::axis_3_point() const { return &pointList_[3]; }
   const double *CoordinateFrame::plane_1_3_point() const { return &pointList_[6]; }
+
+  bool Ioss::CoordinateFrame::operator==(const Ioss::CoordinateFrame &rhs) const
+  {
+    if( this->id_ != rhs.id_ ) {
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.C:29]: Sorry, all output should be via lib::fmt.  Also, why is `operator==` outputting anything anyway...?

File: packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.h [hunk 8]
```diff
@@ -9,6 +9,7 @@
 
 #include <cstdint> // for int64_t
 #include <vector>  // for vector
+#include <stdio.h>   // for printf
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.h:12]: No, not needed in the header file.

File: packages/seacas/libraries/ioss/src/Ioss_ElementTopology.C [hunk 9]
```diff
@@ -311,3 +311,25 @@ Ioss::ElementTopology *Ioss::ElementTopology::boundary_type(int bnd_number) cons
   }
   return nullptr;
 }
+
+bool Ioss::ElementTopology::operator==(const Ioss::ElementTopology &rhs)
+{
+  if( this->name_.compare(rhs.name_) != 0 ) {
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_ElementTopology.C:317]: Operator== should not be outputting anything.

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C [hunk 10]
```diff
@@ -79,3 +79,44 @@ Ioss::Property Ioss::EntityBlock::get_implicit_property(const std::string &my_na
...
+  }
+
+  return Ioss::GroupingEntity::equal( rhs );
+}
+
+bool Ioss::EntityBlock::operator==(const Ioss::EntityBlock &rhs) const
+{
+  /* COMPARE element topologies */
+  if( *(this->topology_) != *(rhs.topology_) ) {
+    printf("EntityBlock: TOPOLOGY mismatch\n");
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C:101]: Do not output stuff from operator==

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C [hunk 11]
```diff
@@ -79,3 +79,44 @@ Ioss::Property Ioss::EntityBlock::get_implicit_property(const std::string &my_na
 
   return Ioss::GroupingEntity::get_implicit_property(my_name);
 }
+
+bool Ioss::EntityBlock::equal(const Ioss::EntityBlock &rhs)
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C:83]: Why is this needed?

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.h [hunk 12]
```diff
@@ -88,6 +88,12 @@ namespace Ioss {
      */
     size_t get_offset() const { return idOffset; }
 
+    // Quiet comparison
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_EntityBlock.h:91]: I think I need an explanation of why we have `equal` and `operator==` and why there is a quiet one and one that outputs the differences.

File: packages/seacas/libraries/ioss/src/Ioss_Field.C [hunk 13]
```diff
@@ -239,6 +239,47 @@ bool Ioss::Field::transform(void *data)
   return true;
 }
 
+bool Ioss::Field::operator==(Ioss::Field rhs)
+{
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_Field.C:243]: Again, please tell me why these were added and why they output information.

File: packages/seacas/libraries/ioss/src/Ioss_GroupingEntity.C [hunk 14]
```diff
@@ -327,3 +327,165 @@ void Ioss::GroupingEntity::property_update(const std::string &property,
     nge->property_add(Ioss::Property(property, value));
   }
 }
+
+bool Ioss::GroupingEntity::equal(const Ioss::GroupingEntity &rhs)
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_GroupingEntity.C:331]: Why?

File: packages/seacas/libraries/ioss/src/Ioss_SideSet.C [hunk 15]
```diff
@@ -170,3 +170,49 @@ void Ioss::SideSet::block_membership(std::vector<std::string> &block_members)
   }
   block_members = blockMembership;
 }
+
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_SideSet.C:173]: Even if both `equal` and `operator==` are needed, we shouldn't be duplicating code between them.

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C [hunk 16]
```diff
@@ -2006,6 +2049,605 @@ void Ioss::Utils::copy_database(Ioss::Region &region, Ioss::Region &output_regio
   }
 }
 
+bool Ioss::Utils::compare_database(Ioss::Region &input_region, Ioss::Region &output_region,
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_Utils.C:2052]: What is this needed for?

File: packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.C [hunk 17]
```diff
@@ -2356,11 +2356,13 @@ int64_t ParallelDatabaseIO::get_field_internal(const Ioss::ElementSet *ns, const
 }
 
 int64_t ParallelDatabaseIO::get_field_internal(const Ioss::SideSet *fs, const Ioss::Field &field,
-                                               void * /* data */, size_t data_size) const
+                                               void *data, size_t data_size) const
 {
   size_t num_to_get = field.verify(data_size);
   if (field.get_name() == "ids") {
     // Do nothing, just handles an idiosyncrasy of the GroupingEntity
+    // However, make sure that the caller gets a consistent answer, i.e., don't leave the buffer full of junk
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.C:2364]: I'll have to think about this...  buffer full of junk is faster; I need to look at when this is called and why...

File: packages/seacas/libraries/ioss/src/faodel/CMakeLists.txt [hunk 18]
```diff
@@ -0,0 +1,87 @@
+# @HEADER
+# ************************************************************************
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/faodel/CMakeLists.txt:2]: Wrong copyright notice and wrong date.

**Review by gsjaardema — CHANGES_REQUESTED** [review]
Thanks for preparing this PR.  Looks good in general, but I think it has a few different concerns all mixed in together.

There are several comments/questions.  A couple pervasive ones:

* Although the `compare` functionality may be useful at some point, I don't understand why it is included in this commit which should be limited to putting Faodel in in place of Data Warehouse.  The changes from compare seem to be  the majority of the changes in the PR.  Is `compare` somehow needed to put FAODEL in?
* I don't want `operator==` outputting any information via printf or other output means.
* If `equal` is how you want to do the output now done in `operator==`, then use `fmt::` instead of `printf`.  Also make sure the output goes to the correct output stream since Sierra clients need control of output.
* Not sure why all the changes to add `TARGET_INCLUDE_DIRECTORIES` -- is that required for FAODEL? If not, should be a separate PR with justification as to why needed.

**Comment by tkordenbrock** [comment]
@gsjaardema Thanks for looking at this PR.  I have pushed updates to resolve your concerns.

>There are several comments/questions. A couple pervasive ones:
>
>   * Although the compare functionality may be useful at some point, I don't understand why it is included in this commit which should be limited to putting Faodel in in place of Data Warehouse. The changes from compare seem to be the majority of the changes in the PR. Is compare somehow needed to put FAODEL in?

The io_shell compare feature was added to aid in the development and testing of the Faodel DatabaseIO component, but it is not required.  I have removed (or split) the commits that add the io_shell compare feature and put those in a separate branch.  I will create a new PR for that branch.

>   * I don't want operator== outputting any information via printf or other output means.
>   * If equal is how you want to do the output now done in operator==, then use fmt:: instead of printf. Also make sure the output goes to the correct output stream since Sierra clients need control of output.

These two issues have been resolved in the io_shell compare feature branch.

>   * Not sure why all the changes to add TARGET_INCLUDE_DIRECTORIES -- is that required for FAODEL? If not, should be a separate PR with justification as to why needed.

The calls to TARGET_INCLUDE_DIRECTORIES() were added to address an installation issue that we found when trying to configure our examples against a standalone SEACAS installation.  It is not required by the Faodel DatabaseIO component, so the commit has been removed.

I will address the inline comments individually.

File: TPL/faodel/runcmake.sh [hunk 19]
```diff
@@ -0,0 +1,91 @@
...
+    LD_EXT="dylib"
+  else
+    LD_EXT="so"
+  fi
+else
+  LD_EXT="a"
+  EXTRA_DEPS="-DNC_EXTRA_DEPS=-ldl\;-lz"
+fi
+
+MPI="${MPI:-OFF}"
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply TPL/faodel/runcmake.sh]: All the Faodel related ON/OFF options have been updated to be YES/NO.

File: TPL/faodel/runcmake.sh [hunk 20]
```diff
@@ -0,0 +1,91 @@
...
+  then
+      export CC=xlc
+      export CXX=xlc
+  fi
+
+fi
+
+if [ "X$BOOST_ROOT" == "X" ] ; then
+  BOOST_ROOT=$(cd ../../../..; pwd)
+  echo "Faodel requires Boost for:" atomic log log_setup serialization
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply TPL/faodel/runcmake.sh]: I have deduped those lines.

File: TPL/kokkos/runcmake.sh [hunk 21]
```diff
@@ -27,8 +27,8 @@ SHARED="${SHARED:-YES}"
 if [ "$MPI" == "YES" ] && [ "$CRAY" = "YES" ]
 then
   export CC=cc
-  export CXX=cxx
-elif [ "$MPI" == "YES" ]
+  export CXX=CC
+elif [ "$MPI" == "ON" ]
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply TPL/kokkos/runcmake.sh]: Fixed in commit f54a315.

File: cmake-config [hunk 22]
```diff
@@ -333,13 +333,14 @@ then
 		    -DTPL_Kokkos_LIBRARIES=${KOKKOS_PATH}/lib/libkokkos.${LD_EXT}"
 fi
 
-if [ "${HAVE_DATAWAREHOUSE}" == "YES" ]
+if [ "$HAVE_FAODEL" == "ON" ]
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply cmake-config]: Fixed in commit f54a315.

File: packages/seacas/libraries/ioss/src/Ioss_GroupingEntity.C (hunk 14 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_GroupingEntity.C]: Moved to branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_SideSet.C (hunk 15 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_SideSet.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C (hunk 16 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_Utils.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.C (hunk 17 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/exodus/Ioex_ParallelDatabaseIO.C:2364]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/faodel/CMakeLists.txt (hunk 18 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/faodel/CMakeLists.txt]: I removed the copyright notice to match the CMakeLists.txt files of the other DatabaseIO plugins (commit e297a3282248685453f21807a0cc4638c8ff9be3).

**Comment by tkordenbrock** [comment]
Looks like github has removed (or hidden) your comment about moving `disable_if_no_fortran()` to `cmake/ProjectDependencies.cmake`.  It was moved to resolve an installation issue similar to the TARGET_INCLUDE_DIRECTORIES() issue.  It isn't related to the Faodel DatabaseIO component so I have reversed it (commit 4e377de125d57d1529a787bddd61a58a6321ecc1).

File: TPLsList.cmake [hunk 23]
```diff
@@ -70,5 +69,7 @@ SET( SEACASProj_TPLS_FINDMODS_CLASSIFICATIONS
   CUDA            "cmake/TPLs/"    SS
   fmt             "cmake/TPLs/"    SS
   Kokkos          "${${PROJECT_NAME}_TRIBITS_DIR}/common_tpls/"  SS
+  Faodel          "${SEACAS_SOURCE_DIR}/cmake/tpls/"    SS
+  Cereal          "${SEACAS_SOURCE_DIR}/cmake/tpls/"    SS
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply TPLsList.cmake:73]: [Cereal](https://uscilab.github.io/cereal/) is a header-only serialization library.  The Faodel DatabaseIO component needs to serialize objects to send across the network.

File: install-tpl.sh (hunk 1 above)
Comment (tkordenbrock) [reply install-tpl.sh]: Fixed in commit f54a315.

File: install-tpl.sh (hunk 2 above)
Comment (tkordenbrock) [reply install-tpl.sh]: Fixed in commit f54a315.

File: packages/seacas/libraries/ioss/src/Ioss_CodeTypes.h (hunk 6 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_CodeTypes.h]: Fixed in commit 7ac8f39.

File: packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.C (hunk 7 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.h (hunk 8 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_CoordinateFrame.h]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_ElementTopology.C (hunk 9 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_ElementTopology.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C (hunk 10 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C (hunk 11 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_EntityBlock.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_EntityBlock.h (hunk 12 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_EntityBlock.h]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/ioss/src/Ioss_Field.C (hunk 13 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_Field.C]: Addressed in branch `tkordenbrock-seacas/add-ioshell-compare`.

File: packages/seacas/libraries/exodus/CMakeLists.txt (hunk 4 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/exodus/CMakeLists.txt]: See https://github.com/gsjaardema/seacas/pull/214#issuecomment-776156905

File: packages/seacas/libraries/ioss/src/CMakeLists.txt (hunk 5 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/CMakeLists.txt]: See https://github.com/gsjaardema/seacas/pull/214#issuecomment-776156905

File: cmake/ProjectDependenciesSetup.cmake (hunk 3 above)
Comment (tkordenbrock) [reply cmake/ProjectDependenciesSetup.cmake]: See https://github.com/gsjaardema/seacas/pull/214#issuecomment-776169266

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C [hunk 24]
```diff
@@ -2257,7 +2257,7 @@ namespace {
       for (const auto &set : sets) {
         const std::string &name = set->name();
         if (options.debug && rank == 0) {
-          fmt::print(Ioss::DEBUG(), "{}, ", name);
+          fmt::print(stderr, "{}, ", name);
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_Utils.C:2260]: The 'stderr' should be 'Ioss::DEBUG()'

Similar for some locations down below.

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C [hunk 25]
```diff
@@ -2436,13 +2445,16 @@ namespace {
...
       // generates overhead...
+
       if (field_name == "connectivity" && ige->type() != Ioss::ELEMENTBLOCK) {
         continue;
       }
       if (field_name == "ids") {
         continue;
       }
-      if (Ioss::Utils::substr_equal(prefix, field_name)) {
+
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/Ioss_Utils.C:2455]: Is the `Ioss::Utils::substr_equal()` not valid here?

File: packages/seacas/libraries/ioss/src/init/CMakeLists.txt [hunk 26]
```diff
@@ -39,9 +39,9 @@ TRIBITS_ADD_LIBRARY(
 	Ionit
 	HEADERS	${HEADERS}
 	SOURCES ${SOURCES}
-	DEPLIBS Ioss ${EXODUSLIB}
-	        Iogn Iogs Iohb Iotr ${IOVISLIB}
-            ${PAMGENLIB} ${CGNSLIB} ${VISLIB} ${DATAWAREHOUSELIB} ${ADIOS2LIB}
+	DEPLIBS Ioss ${EXODUSLIB} ${PAREXOLIB}
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/init/CMakeLists.txt:42]: There is no need for `PAREXOLIB` or `EXODUSFAC` anymore.  There is a single `EXODUSLIB`

File: packages/seacas/libraries/ioss/src/visualization/catalyst/applications/ioss2catalyst/IossApplication.cxx [hunk 27]
```diff
@@ -0,0 +1,755 @@
+// Copyright(C) 1999-2020 National Technology & Engineering Solutions
```
Comment (gsjaardema) [inline packages/seacas/libraries/ioss/src/visualization/catalyst/applications/ioss2catalyst/IossApplication.cxx:1]: Is this part of the FAODEL changes, or did it somehow get accidentally added?

**Review by gsjaardema — CHANGES_REQUESTED** [review]
Looking much better.  A couple requested changes and then it should be ready to merge.

File: cmake-travis.sh [hunk 28]
```diff
@@ -50,7 +50,7 @@ function check_enable()
 HAVE_NETCDF=$(check_enable "${NETCDF_PATH}/include/netcdf.h")
 HAVE_MATIO=$(check_enable "${MATIO_PATH}/include/matio.h")
 HAVE_CGNS=$(check_enable "${CGNS_PATH}/include/cgnslib.h")
-HAVE_DATAWAREHOUSE=OFF
+HAVE_FAODEL=$(check_enable "${CGNS_PATH}/include/faodel/faodelConfig.h")
```
[the remark this answers is not in the document]
Comment (tkordenbrock) [reply cmake-travis.sh]: Fixed in 778cc92.

File: packages/seacas/libraries/ioss/src/init/CMakeLists.txt (hunk 26 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/init/CMakeLists.txt]: Fixed in 778cc92.

File: packages/seacas/libraries/ioss/src/visualization/catalyst/applications/ioss2catalyst/IossApplication.cxx (hunk 27 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/visualization/catalyst/applications/ioss2catalyst/IossApplication.cxx]: This is another part of the app integration code.  It has been removed (commit 77f7afb).  We can add it back later under a different PR.

**Review by gsjaardema — APPROVED** [review]

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C (hunk 25 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_Utils.C]: This is a cherry pick merge error.  I fixed it in 135650d.

File: packages/seacas/libraries/ioss/src/Ioss_Utils.C (hunk 24 above)
Comment (tkordenbrock) [reply packages/seacas/libraries/ioss/src/Ioss_Utils.C]: This is another cherry pick merge error.  I fixed these plus two other "stderr" uses that I found in `transfer_assemblies()` and `transfer_blobs()` (commit 135650d).

**Comment by gsjaardema** [comment]
Thanks.  Sorry for all the comments.
